api/internal/clients/kube.go

```diff
@@ -0,0 +1,89 @@
+package clients
+
+import (
+	"fmt"
+
+	autopilotv1beta2 "github.com/k0sproject/k0s/pkg/apis/autopilot/v1beta2"
+	k0shelmv1beta1 "github.com/k0sproject/k0s/pkg/apis/helm/v1beta1"
+	k0sv1beta1 "github.com/k0sproject/k0s/pkg/apis/k0s/v1beta1"
+	embeddedclusterv1beta1 "github.com/replicatedhq/embedded-cluster/kinds/apis/v1beta1"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/metadata"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var schemeBuilder = runtime.NewSchemeBuilder(
+	corev1.AddToScheme,
+	embeddedclusterv1beta1.AddToScheme,
+	//nolint:staticcheck // SA1019 we are using the deprecated scheme for backwards compatibility, we can remove this once we stop supporting k0s v1.30
+	autopilotv1beta2.AddToScheme,
+	//nolint:staticcheck // SA1019 we are using the deprecated scheme for backwards compatibility, we can remove this once we stop supporting k0s v1.30
+	k0sv1beta1.AddToScheme,
+	//nolint:staticcheck // SA1019 we are using the deprecated scheme for backwards compatibility, we can remove this once we stop supporting k0s v1.30
+	k0shelmv1beta1.AddToScheme,
+	velerov1.AddToScheme,
+)
+
+type KubeClientOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+	KubeConfigPath   string
+}
+
+func getScheme() *runtime.Scheme {
+	s := runtime.NewScheme()
+	schemeBuilder.AddToScheme(s)
+	return s
+}
```

Comment:
Thoughts on using this VS the global `scheme.Scheme`? Personally I'm not super fond of the global/init() pattern but would be curious to hear y'alls thoughts.